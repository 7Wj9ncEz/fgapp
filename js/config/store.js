import { createStore, combineReducers } from 'redux';

import articlesReducer from '../reducers/articles-reducer';
import professorsReducer from '../reducers/professors-reducer';

export const initialState = {
  articles: {
    page: 1,
    lastPage: Infinity,
    data: [],
    error: false
  },
  professors: []
};


const combinedReducers = combineReducers({
  articles: articlesReducer,
  professors: professorsReducer
});

const store = createStore(combinedReducers, initialState);

export default store;
