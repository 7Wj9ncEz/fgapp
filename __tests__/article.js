import 'react-native';
import React from 'react';
import Article from '../js/components/article.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import store from './../js/config/store';
import { Provider } from 'react-redux';

it('renders correctly', () => {
  const tree = renderer.create(
    <Article
      title={"lala"}
      body={"<img src=\"/unb-gama/imagens/imagens.png\" />"}
      date={"2016/04/04 10:10:10"}
      authorName={"vitor"}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('parses correctly', () => {
  let article = new Article({title: "lala", body: "<img src=\"/unb-gama/imagens/imagens.png\" />", date: "2016/04/04 10:10:10", authorName: "vitor"});

  expect(article.state.parsedBody).toBe("<img src=\"https://fga.unb.br/unb-gama/imagens/imagens.png\" />");
});
