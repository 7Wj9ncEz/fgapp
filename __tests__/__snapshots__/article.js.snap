exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingLeft": 20,
      "paddingRight": 20,
    }
  }>
  <View
    style={
      Object {
        "flex": 7,
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#005263",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }>
      lala
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 2,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textAlign": "right",
        }
      }>
      04
      /
      04
      /
      2016
       
      vitor
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 17,
      }
    }>
    <View
      style={
        Object {
          "flex": 1,
        }
      }>
      <RCTWebView
        allowsInlineMediaPlayback={undefined}
        automaticallyAdjustContentInsets={undefined}
        bounces={undefined}
        contentInset={undefined}
        dataDetectorTypes={undefined}
        decelerationRate={undefined}
        injectedJavaScript={undefined}
        mediaPlaybackRequiresUserAction={undefined}
        messagingEnabled={false}
        onLoadingError={[Function]}
        onLoadingFinish={[Function]}
        onLoadingStart={[Function]}
        onMessage={[Function]}
        onShouldStartLoadWithRequest={undefined}
        scalesPageToFit={undefined}
        scrollEnabled={undefined}
        source={
          Object {
            "html": "<img src=\"https://fga.unb.br/unb-gama/imagens/imagens.png\" />",
          }
        }
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "#ffffff",
            },
            undefined,
          ]
        } />
    </View>
  </View>
</View>
`;
